AspNetCore.CacheOutput/InvalidateCacheOutputAttribute.cs

```diff
@@ -26,17 +26,27 @@ public InvalidateCacheOutputAttribute(string methodName, Type controllerType, Ty
             this.cacheKeyGeneratorType = cacheKeyGeneratorType;
         }
 
-        public override async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        private bool IsFailure(int statusCode)
         {
-            await base.OnActionExecutionAsync(context, next);
-
-            if (
-                context.HttpContext.Response != null &&
-                !(
-                    context.HttpContext.Response.StatusCode >= (int)HttpStatusCode.OK && 
-                    context.HttpContext.Response.StatusCode < (int)HttpStatusCode.Ambiguous
-                )
-            )
+            return !(
+                    statusCode >= (int)HttpStatusCode.OK &&
+                    statusCode < (int)HttpStatusCode.Ambiguous
+                );
+        }
+
+        public override async Task OnResultExecutionAsync(ResultExecutingContext context, ResultExecutionDelegate next)
+        {
+            await base.OnResultExecutionAsync(context, next);
+
+#if NETCOREAPP2_0 || NETCOREAPP2_1
```

Comment:
.NET Core 2.2 also should be handled.

Reply:
.Net Core 2.2+ is handled by the else condition. 2.0 and 2.1 are the only ones that need to be specially cared for

Reply:
So the IStatusCodeActionResult interface is only available for .Net Core 2.2+.  As a result, I only need to handle the the earlier ones. Going forward, the new code handles the rest.